### PR TITLE
`ingest.py` cleanups, doc nits

### PR DESCRIPTION
**Pull request by ryan-williams**
**Issue and/or context:** while working on #2252, cleaned up a few things while trying to understand/audit them.

### CODE REVIEW

File: apis/python/src/tiledbsoma/io/conversions.py [hunk 1]
```diff
@@ -46,7 +46,7 @@ def to_tiledb_supported_array_type(name: str, x: _MT) -> _MT:
     E.g., float16 -> float32
     """
     if isinstance(x, (np.ndarray, sp.spmatrix)) or not isinstance(
-        x, pd.CategoricalDtype
+        x.dtype, pd.CategoricalDtype
```
Comment (ryan-williams) [inline apis/python/src/tiledbsoma/io/conversions.py:49]: I think this `if` condition was trivially evaluating as `True`, previously ([last commit](https://github.com/single-cell-data/TileDB-SOMA/commit/616823f4697c08db3043319cc68642551aaa44f5#diff-14e9163b10b9e187b5e590445e41f83c891a991bc670b9bda9cf194217532fc4L49-R50)).

File: apis/python/src/tiledbsoma/io/ingest.py [hunk 2]
```diff
@@ -615,136 +573,60 @@ def from_anndata(
...
-                                platform_config=platform_config,
-                                context=context,
-                                axis_0_mapping=jidmaps.obs_axis,
-                                axis_1_mapping=AxisIDMapping.identity(num_cols),
-                            ) as arr:
-                                _maybe_set(
-                                    obsm, key, arr, use_relative_uri=use_relative_uri
-                                )
-                            arr.close()
-                    measurement.obsm.close()
```
Comment (ryan-williams) [inline apis/python/src/tiledbsoma/io/ingest.py:650]: These `close`s don't exist in the `varm`/`obsp`/`varp` blocks below; they seem unnecessary, but let's confirm or refute that.

File: apis/python/src/tiledbsoma/io/ingest.py [hunk 3]
```diff
@@ -615,136 +573,60 @@ def from_anndata(
...
-
-                if len(anndata.varm.keys()) > 0:  # do not create an empty collection
-                    _util.uri_joinpath(measurement_uri, "varm")
```
Comment (ryan-williams) [inline apis/python/src/tiledbsoma/io/ingest.py:653]: no-op?

File: apis/python/src/tiledbsoma/io/ingest.py [hunk 4]
```diff
@@ -615,136 +573,60 @@ def from_anndata(
...
-                            ) as darr:
-                                _maybe_set(
-                                    varm,
-                                    key,
-                                    darr,
-                                    use_relative_uri=use_relative_uri,
-                                )
-
-                if len(anndata.obsp.keys()) > 0:  # do not create an empty collection
-                    _util.uri_joinpath(measurement_uri, "obsp")
```
Comment (ryan-williams) [inline apis/python/src/tiledbsoma/io/ingest.py:688]: no-op?

File: apis/python/src/tiledbsoma/io/ingest.py [hunk 5]
```diff
@@ -615,136 +573,60 @@ def from_anndata(
...
-                            ) as sarr:
-                                _maybe_set(
-                                    obsp,
-                                    key,
-                                    sarr,
-                                    use_relative_uri=use_relative_uri,
-                                )
-
-                if len(anndata.varp.keys()) > 0:  # do not create an empty collection
-                    _util.uri_joinpath(measurement_uri, "obsp")
```
Comment (ryan-williams) [inline apis/python/src/tiledbsoma/io/ingest.py:719]: no-op + copy-pasta (should be `"varp"`)

File: apis/python/src/tiledbsoma/io/ingest.py [hunk 6]
```diff
@@ -615,136 +573,60 @@ def from_anndata(
...
-                                    use_relative_uri=use_relative_uri,
-                                )
+                def _ingest_obs_var_m_p(
+                    ad_key: Literal["obsm", "varm", "obsp", "varp"],
+                    axis_0_mapping: AxisIDMapping,
+                    axis_1_mapping: Optional[AxisIDMapping] = None,
+                ) -> None:
+                    ad_val = getattr(anndata, ad_key)
+                    if len(ad_val.keys()) > 0:  # do not create an empty collection
+                        val_uri = _util.uri_joinpath(measurement_uri, ad_key)
```
Comment (ryan-williams) [inline apis/python/src/tiledbsoma/io/ingest.py:583]: double-check: `measurement_uri` vs. `measurement.uri` here

**Review by ryan-williams — COMMENTED** [review]
Left a couple comments highlighting areas we should look at, @johnkerl (once I get CI passing)

File: apis/python/src/tiledbsoma/io/ingest.py (hunk 6 above)
Comment (johnkerl) [reply apis/python/src/tiledbsoma/io/ingest.py]: @ryan-williams for example
https://github.com/single-cell-data/TileDB-SOMA/blob/1.8.0/apis/python/src/tiledbsoma/io/ingest.py#L510-L517

**Comment by ryan-williams** [comment]
I've rebased this, and removed an `IngestionParams.__init__` refactor (https://github.com/single-cell-data/TileDB-SOMA/commit/39fe37ab) that @johnkerl and I were not quite sold on.

**Pull request merged**